src/scss/mixins/_button.scss

```diff
@@ -44,15 +50,6 @@
         box-shadow: $shadow-focus;
     }
     &:disabled {
-        background-color: $gray400;
-        color: text-contrast($gray400);
-    }
-
-    //in case button has icon
-    span {
-        margin: 0 0.2rem;
-    }
-    i {
-        margin: 0 0.2rem;
+        @include smartColor($gray400);
```

Comment:
**smart color** take one parameter and set his value to background.
and calculate the color luminance and give **dynamic** color.